store/sqlstore/user_store.go

```diff
@@ -1361,3 +1361,466 @@ func (us SqlUserStore) InferSystemInstallDate() store.StoreChannel {
+				        AND TargetInteraction.UserId != :uid
+				        AND Parent_Post_Author_Id = :uid
+				        AND ChannelMembersUsers.Id = :uid) -- just want one user returned from channel members (the current user) (will record as a reply)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, not written by current user, to a post that the current user IS NOT an author of, in a group channel
+				        AND TargetInteraction.UserId != :uid
+				        AND Parent_Post_Author_Id != :uid
+				        AND ChannelMembersUsers.Id = :uid) -- just want one user returned from channel members (the current user) (will record as a DM)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, written by current user, to a post that the current user IS NOT an author of, in a group channel
+				        AND TargetInteraction.UserId = :uid
+				        AND Parent_Post_Author_Id != :uid
+				        AND ChannelMembersUsers.Id != :uid)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, written by current user, to a post that the current user IS an author of, in a group channel
+				        AND TargetInteraction.UserId = :uid
+				        AND Parent_Post_Author_Id = :uid
+				        AND ChannelMembersUsers.Id != :uid)
+				)
+			) AS UserInteractions
+
+			WHERE Username IN (SELECT Username -- all interactions must take place with users on your team
+			                          FROM Users
+
+			                            JOIN TeamMembers
+			                            ON TeamMembers.TeamID = :teamId
+			                            AND TeamMembers.UserId = Users.Id
+			                        )
+			`
+			var userInteractions []*model.UserInteraction
+
+			if _, err :=  us.GetReplica().Select(&userInteractions, query, map[string]interface{}{"uid": uid, "teamId": teamId}); err != nil {
+				result.Err = model.NewAppError("SqlUserStore.GetUserInteractions", "store.sql_post.ge_user_interactions.app_error", nil, err.Error(), http.StatusInternalServerError)
+			}
+
+			result.Data = userInteractions
+	})
+}
+
+func (us *SqlUserStore) GetUserLearningGroups(uid string, teamId string, learningGroupPrefixes []*model.LearningGroupPrefix) store.StoreChannel {
+	// For every learning group prefix for this team, use this case statement
+	// to record a name for this channel(team), minus the prefix (ex. plg-30 => 30)
+	prefixReplaceQuery := ""
+
+	for _, prefix := range learningGroupPrefixes {
+		prefixReplaceQuery += `
+		WHEN LEFT(Channels.Name,` + strconv.Itoa(len(prefix.Prefix) + 1) + `) = '` + prefix.Prefix + `-'
+		THEN REPLACE(Channels.Name,'` + prefix.Prefix + `-','')
+		`
+	}
+
+	// If there are no learning group prefixes, then '' as team name,
+	// since this should be a 1 to 1 relationship
+	if len(prefixReplaceQuery) == 0 {
+		prefixReplaceQuery = " WHEN 1 = 1 THEN '' "
+	}
+
+	// For every learning group prefix for this team, use this case statement
```

Comment:
This is _very_ similar to the loop above. The comment should document what is different. It looks to me like it's just the final line. Again, the boilerplate here suggests you could abstract this loop out as a function and just call it in the 3 places it's being used, along with the associated if.

Also, StringBuilder is looking increasingly like a good idea, since we're doing this quadratic-time string construction in so many places.